From 51f2372df1d1df7a5a160999ba1ed7d1bba51e86 Mon Sep 17 00:00:00 2001
From: jinghua <xigu@marvell.com>
Date: Thu, 13 Apr 2017 20:55:05 +0800
Subject: [PATCH 0974/1345] doc: a3700: update doc for usb static host mode

commit  d2ba6af2b19d9356a98dc51fcdcd972044696d87 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

- A3700 used to have to use two different compatible string for USB3/2
  host controller:
  - For USB static host mode: "xhci-plat".
  - For OTG mode: "marvell,armada-3700-xhci", in this case, xhci driver
    does not add hcd to the bus, it registers itself to OTG PHY driver,
    and when OTG PHY detects USB DOK plugged-in, it will register the
    hcd to the bus.
- Now since OTG PHY driver has already supported usb static host mode,
  xhci driver could always uses "marvell,armada-3700-xhci", and the
  OTG PHY driver will take care of everything.

Change-Id: Ia7c1b01bd19bce5ed1943c4a6b546490bf02f22f
Signed-off-by: jinghua <xigu@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/38675
Reviewed-by: Hua Jing <jinghua@marvell.com>
Tested-by: Hua Jing <jinghua@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/38755
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../devicetree/bindings/usb/armada3700-usb-phy.txt |   11 +----------
 Documentation/devicetree/bindings/usb/usb-xhci.txt |    4 ++--
 2 files changed, 3 insertions(+), 12 deletions(-)

diff --git a/Documentation/devicetree/bindings/usb/armada3700-usb-phy.txt b/Documentation/devicetree/bindings/usb/armada3700-usb-phy.txt
index 9e3ed39..e83f6ad 100644
--- a/Documentation/devicetree/bindings/usb/armada3700-usb-phy.txt
+++ b/Documentation/devicetree/bindings/usb/armada3700-usb-phy.txt
@@ -5,9 +5,6 @@ This controller is integrated only in Armada-3700.
 Armada-3700 could be working in USB OTG mode which supports USB Host/Device mode
 switching dynamically. To do that, the following dts changes are required:
 - Enable both USB Host (usb3) and Device (u3d and udc) controller DT nodes.
-- In USB Host DT node, use special XHCI driver compatible string:
-  "marvell,armada-3700-xhci-otg", while the default one "generic-xhci" is for
-  USB static host mode.
 - Enable OTG PHY TD node and make sure XHCI DT node have reference to it with
   "usb-phy" property.
 Armada-3700 could also be working in USB static host mode, when USB Device feature
@@ -24,12 +21,6 @@ Optional properties:
   switching supported.
 
 USB OTG mode example:
-	usb3@58000 {
-		compatible = "marvell,armada-3700-xhci";
-		status = "okay";
-		usb-phy = <&otg_phy>;
-	};
-
 	usb_phy: usb_phy@5d800 {
 		compatible = "marvell,armada-3700-usb-phy";
 		reg = <0x5d800 0x100>;
@@ -38,7 +29,7 @@ USB OTG mode example:
 		status = "okay";
 	};
 
-USB Static Host mode example, no modification to USB3 host controller DT node:
+USB Static Host mode example:
 	usb_phy: usb_phy@5d800 {
 		compatible = "marvell,armada-3700-usb-phy";
 		reg = <0x5d800 0x100>;
diff --git a/Documentation/devicetree/bindings/usb/usb-xhci.txt b/Documentation/devicetree/bindings/usb/usb-xhci.txt
index 0418546..743e30d 100644
--- a/Documentation/devicetree/bindings/usb/usb-xhci.txt
+++ b/Documentation/devicetree/bindings/usb/usb-xhci.txt
@@ -48,11 +48,11 @@ Examples:
 
 	For Armada 3700 which uses separated PHYs for USB 2 and USB 3:
 	usb3@58000 {
-		compatible = "generic-xhci";
+		compatible = "marvell,armada-3700-xhci";
 		reg = <0x58000 0x4000>;
 		interrupts = <GIC_SPI 3 IRQ_TYPE_LEVEL_HIGH>;
 		clocks = <&sgateclk 17>;
-		usb-phy = <&usb3_phy>;
+		usb-phy = <&usb_phy>;
 		phys = <&utmi_usb32>, /* utmi usb32 dedicated phy */
 		       <&a3700_comphy 1 COMPHY_USB3>; /* usb3 comphy */
 		phy-names = "usb2", "usb3";
-- 
1.7.9.5

